test/e2e-png.js

```diff
@@ -11,12 +11,16 @@ const fs             = require("fs");
 const fsP            = pify(fs);
 const sourceFile     = `${__dirname}/fixture/events/s3_put_file.json`;
 const setting        = JSON.parse(fs.readFileSync(sourceFile));
+const AWS            = require("aws-sdk-mock");
 
 let processor;
 let images;
 let fileSystem;
 
 test.before(async t => {
+    AWS.mock("S3", "deleteObject", (params, callback) => {
```

Comment:
Yes, I think that we have to mocking `fileSystem.deleteObject()` because `sinon.spy()` spies target function but original target function will called. Just in case, simply I want to confirm `deleteObject()` was called in `ImageProcessor`, but this method reject Promise with `Access Denied()`, so I used mock and spy. If you have any good idea, please let me know, thanks.